### PR TITLE
[FeatureStore] Fix Redis server-type config and limit pandas to <1.5.0

### PR DESCRIPTION
**Pull request by assaf758**
- fix propagation of redis configs to runtimes.
- WA for [ML-2644](https://jira.iguazeng.com/browse/ML-2644) - limiting pandas to <1.5.0

### CODE REVIEW

File: mlrun/config.py [hunk 1]
```diff
@@ -114,6 +114,7 @@
     "v3io_api": "http://v3io-webapi:8081",
     "redis": {
         "url": "",
+        "type": "",
```
Comment (Tankilevitch) [inline mlrun/config.py:117]: add comment of supported redis types which points to the class you created

Comment (Tankilevitch) [reply mlrun/config.py:117]: also is this param required? or can we infer it by some way?

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -312,7 +312,8 @@ def connect(self, secrets=None):
                 else server_cfg.get("generate_artifact_target_path_from_artifact_hash")
             )
 
-            config.redis.url = config.redis.url or server_cfg.get("redis.url")
+            config.redis.url = config.redis.url or server_cfg.get("redis_url")
+            config.redis.type = config.redis.type or server_cfg.get("redis_type")
```
Comment (Tankilevitch) [inline mlrun/db/httpdb.py:316]: add comment why you prefer the client to have precedence over the server.

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, I think you are missing a test which check initialization of possible redis types, just to make sure the code behaves as expected.
Also lets add a bit more documentation about the purpose of specifying a type.

File: mlrun/config.py (hunk 1 above)
Comment (assaf758) [reply mlrun/config.py]: > add comment of supported redis types which points to the class you created

added comment.

> also is this param required? or can we infer it by some way?

in theory we can try one type, and if does not work (we get an error) try the other type. i don't think we want this.

File: mlrun/db/httpdb.py (hunk 2 above)
Comment (assaf758) [reply mlrun/db/httpdb.py]: added

**Comment by assaf758** [comment]
>Looks good, I think you are missing a test which check initialization of possible redis types, just to make sure the code behaves as expected.

unfortunately I currently don't have a redis cluster to test against. I was able to manually test that the redis-type has an effect.

File: mlrun/db/httpdb.py [hunk 3]
```diff
@@ -312,7 +312,9 @@ def connect(self, secrets=None):
                 else server_cfg.get("generate_artifact_target_path_from_artifact_hash")
             )
 
-            config.redis.url = config.redis.url or server_cfg.get("redis.url")
+            config.redis.url = config.redis.url or server_cfg.get("redis_url")
+            # allow client to set the default partial WA for luck of support of per-target auxiliary options
```
Comment (quaark) [inline mlrun/db/httpdb.py:316]:
```suggestion
            # allow client to set the default partial WA for lack of support of per-target auxiliary options
```

**Review by quaark — CHANGES_REQUESTED** [review]
LGTM. Just a minor spelling comment for clarity

File: requirements.txt [hunk 4]
```diff
@@ -27,7 +27,8 @@ ipython~=7.0
 nuclio-jupyter~=0.9.1
 # >=1.16.5 from pandas 1.2.1 and <1.23.0 from storey
 numpy>=1.16.5, <1.23.0
-pandas~=1.2
+# limiting pandas to <1.5.0 since 1.5.0 causes exception in storey on casting from ns to us
+pandas~=1.2, <1.5.0
```
Comment (Tankilevitch) [inline requirements.txt:31]:
```suggestion
pandas~=1.2.0
```

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
please rename title accordingly

File: requirements.txt (hunk 4 above)
Comment (quaark) [reply requirements.txt:31]: This will pin to 1.2.x while we can safely go up to 1.4.x, no?
I think @assaf758 did it right

**Pull request merged**